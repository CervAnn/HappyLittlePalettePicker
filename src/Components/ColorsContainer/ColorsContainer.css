.ColorsContainer{
  display: flex;
  flex-flow: row wrap;
  float: right;
  width: 90%;
  margin-bottom: 20px;
  border-radius: 20px;
  font-family: 'Caveat', cursive;
}

#paletteName-container {
  width: 100%;
  color: white;
  font-size: 30px;
  margin-bottom: 15px;
  text-shadow: 
  -1px -1px 0 #000, 
  1px -1px 0 #000, 
  -1px 1px 0 #000,
  1px 1px 0 #000;
}

