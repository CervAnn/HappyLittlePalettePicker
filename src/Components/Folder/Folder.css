.Folder {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  margin-bottom: 5px;
}

#folder_name {
  border: none;
  background-color: transparent;
  width: 75%;
  height: 50px;
  text-align: center;
  align-self: center;
  font-size: 30px;
  font-weight: 800;
  color: white;
}

#folder_name:hover {
  background-color: white;
  color: #000;
}

#delete-img {
  height: 25px;
  margin: 2px 2px 2px 2px;
}

#delete-img:hover {
  border: 1px solid #FFFFFF;
}

/* .Folder {
  display: flex;
  font-size: 16px;
}

.folder_delete {
  height: 25px;
  margin: 10px;
}
.folder_delete:hover {
  filter: invert(0.5)
}

.folder_more {
  height: 25px;
  margin: 10px;
}

.folder_more:hover {
  filter: invert(0.5)
}

#folder_name {
  margin: 10px;
  font-size: 26px;
}

#folder_name:hover {
  color: grey
} */