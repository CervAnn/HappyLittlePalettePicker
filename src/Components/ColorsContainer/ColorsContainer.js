import React from 'react'
import ColorCard from '../ColorCard/ColorCard'
import './ColorsContainer.css'
import PropTypes from 'prop-types';

const ColorsContainer = (props) => {
  const allColors = props.currentPalette.map((swatch, index) => {
    return (
      <ColorCard
      name={swatch.name ? <h2 id="savedPaletteName">{swatch.name}</h2> : undefined}
      delete={props.deleteAPalette || null}
      folderID={props.folderID || null}
      key={index}
      color={swatch.color}
      isLocked={swatch.isLocked}
      toggleLock={props.toggleLock}
      id={swatch.id}
      />
    )
  })

  return (
    <section className='ColorsContainer'>
      {props.paletteName ? 
      <article id="paletteName-container">
        <h2 id="savedPaletteName">{props.paletteName}</h2>
        <img id="delete-palette-img" 
        src={require('../../assets/delete.svg')} 
        onClick={(e) => props.deleteAPalette(e, props.id, props.folderID)}/>
      </article>
      : 
      undefined}
      {allColors}
    </section>
  )
}

export default ColorsContainer;

ColorsContainer.propTypes = {
  currentPalette: PropTypes.array,
  folderID: PropTypes.string,
  folders: PropTypes.object,
  palettes: PropTypes.array,
  toggleLock: PropTypes.func
} 