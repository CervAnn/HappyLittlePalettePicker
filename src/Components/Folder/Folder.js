import React from 'react'
import './Folder.css'

const Folder = (props) => {
  return (
    <article className="Folder">
      <button 
      onClick={(e) => props.getAFoldersPalettes(e, props.id)}
      id="folder_name">{props.name}
      </button>
      <img 
      id="delete-img" 
      src={require('../../assets/delete.svg')}
      onClick={(e) => props.deleteAFolder(e, props.id)}
      />
    </article>
  )
}

export default Folder
