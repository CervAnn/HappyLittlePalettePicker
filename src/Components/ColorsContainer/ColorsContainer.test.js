import React from 'react';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import { configure } from 'enzyme';
import Folder from '../../Components/Folder/Folder'
import ColorsContainer from './ColorsContainer'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ColorsContainer', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<ColorsContainer
    currentPalette={[
      {color: "#2C7559", isLocked: false}, 
      {color: "57d576", isLocked: false}, 
      {color: "#a35c95", isLocked: false}, 
      {color: "#461d33", isLocked: false}, 
      {color: "413134", isLocked: false}
    ]}
    deleteAPalette={jest.fn()}
    folderID={16558}
    id={114}
    name={"Whale Life"}
    />);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call deleteAPalette on a click', () => {
    let palettes = [{name: "palette1", c1: "#FFFFFF", c2: "#FFFFFF", c3: "#FFFFFF", c4: "#FFFFFF", c5: "#FFFFFF", folder_id: 33}, {name: "palette2", c1: "#FFFFFF", c2: "#FFFFFF", c3: "#FFFFFF", c4: "#FFFFFF", c5: "#FFFFFF", folder_id: 33}]
    const wrapperA = shallow(<Folder key="1" id="2" name={name} palettes={palettes} getAFoldersPalettes={jest.fn()} deleteAFolder={jest.fn()} onClick={jest.fn()}/>)

    const mockEvent = { preventDefault: jest.fn() }

    wrapperA.find('#folder_name').simulate('click', mockEvent);

    const mockDeleteAPalette = jest.fn()

    const wrapper = shallow(<ColorsContainer
      currentPalette={[
        {color: "#2C7559", isLocked: false}, 
        {color: "57d576", isLocked: false}, 
        {color: "#a35c95", isLocked: false}, 
        {color: "#461d33", isLocked: false}, 
        {color: "413134", isLocked: false}
      ]}
      deleteAPalette={mockDeleteAPalette}
      folderID={16558}
      id={114}
      paletteName={"Whale Life"}
      onClick={jest.fn()}
      />);

    wrapper.find('img').simulate('click', mockEvent);

    expect(mockDeleteAPalette).toHaveBeenCalledWith(mockEvent, 114, 16558);
  });

  it('should test that a palette name does not render if there is no name for the palette', () => {

    const wrapper = shallow(<ColorsContainer
      currentPalette={[
        {color: "#2C7559", isLocked: false}, 
        {color: "57d576", isLocked: false}, 
        {color: "#a35c95", isLocked: false}, 
        {color: "#461d33", isLocked: false}, 
        {color: "413134", isLocked: false}
      ]}
      folderID={""}
      folders={{ folders: [
        {created_at: "2019-10-16", id: 16556, name: "Bold", updated_at: "2019-10-17"}, 
        {created_at: "2019-10-16", id: 16557, name: "Shy", updated_at: "2019-10-17"}, 
        {created_at: "2019-10-16", id: 16558, name: "Animals", updated_at: "2019-10-17"}, 
        {created_at: "2019-10-16", id: 16559, name: "Winding Down", updated_at: "2019-10-17"}, 
        {created_at: "2019-10-16", id: 16560, name: "Monster Mash", updated_at: "2019-10-17"}
      ]}}
      palettes={[]}
      toggleLock={jest.fn()}
      />);
      const name = wrapper.find('h2');
      expect(name).toHaveLength(0);
    });

  it('should test that a palette name does render if there is a name for the palette', () => {
    const wrapper = shallow(<ColorsContainer
      currentPalette={[
        {color: "#2C7559", isLocked: false}, 
        {color: "57d576", isLocked: false}, 
        {color: "#a35c95", isLocked: false}, 
        {color: "#461d33", isLocked: false}, 
        {color: "413134", isLocked: false}
      ]}
      deleteAPalette={jest.fn()}
      folderID={16558}
      id={114}
      paletteName={"Whale Life"}
      onClick={jest.fn()}
      />);
    const name = wrapper.find('h2')
    expect(name).toHaveLength(1);
    // expect(mockDeleteAPalette).toHaveBeenCalledWith(mockEvent, 114, 16558);
  });
})