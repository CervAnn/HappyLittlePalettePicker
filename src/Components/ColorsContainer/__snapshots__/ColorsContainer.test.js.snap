// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorsContainer should match the snapshot with all data passed in correctly 1`] = `
<section
  className="ColorsContainer"
>
  <ColorCard
    color="#2C7559"
    delete={[MockFunction]}
    folderID={16558}
    isLocked={false}
    key="0"
  />
  <ColorCard
    color="57d576"
    delete={[MockFunction]}
    folderID={16558}
    isLocked={false}
    key="1"
  />
  <ColorCard
    color="#a35c95"
    delete={[MockFunction]}
    folderID={16558}
    isLocked={false}
    key="2"
  />
  <ColorCard
    color="#461d33"
    delete={[MockFunction]}
    folderID={16558}
    isLocked={false}
    key="3"
  />
  <ColorCard
    color="413134"
    delete={[MockFunction]}
    folderID={16558}
    isLocked={false}
    key="4"
  />
</section>
`;
